great_expectations/datasource/data_connector/configured_asset_azure_data_connector.py

```diff
@@ -0,0 +1,146 @@
+            azure_options (dict): wrapper object for **kwargs
+            batch_spec_passthrough (dict): dictionary with keys that will be added directly to batch_spec
+        """
+        logger.debug(f'Constructing ConfiguredAssetAzureDataConnector "{name}".')
+
+        super().__init__(
+            name=name,
+            datasource_name=datasource_name,
+            execution_engine=execution_engine,
+            assets=assets,
+            default_regex=default_regex,
+            sorters=sorters,
+            batch_spec_passthrough=batch_spec_passthrough,
+        )
+        self._container = container
+        self._name_starts_with = os.path.join(name_starts_with, "")
+        self._delimiter = delimiter
+
+        if azure_options is None:
+            azure_options = {}
+
+        try:  # Both connection string and standard auth methods are supported
+            if "conn_str" in azure_options:
+                self._azure = BlobServiceClient.from_connection_string(**azure_options)
+            else:
+                self._azure = BlobServiceClient(**azure_options)
+        except (TypeError, AttributeError):
+            raise ImportError(
+                "Unable to load Azure BlobServiceClient (it is required for ConfiguredAssetAzureDataConnector)."
+            )
+
+    def build_batch_spec(self, batch_definition: BatchDefinition) -> AzureBatchSpec:
+        """
+        Build BatchSpec from batch_definition by calling DataConnector's build_batch_spec function.
+
+        Args:
+            batch_definition (BatchDefinition): to be used to build batch_spec
+
+        Returns:
+            BatchSpec built from batch_definition
+        """
+        batch_spec: PathBatchSpec = super().build_batch_spec(
+            batch_definition=batch_definition
+        )
+        return AzureBatchSpec(batch_spec)
+
+    def _get_data_reference_list_for_asset(self, asset: Optional[Asset]) -> List[str]:
+        query_options: dict = {
+            "container": self._container,
+            "name_starts_with": self._name_starts_with,
+            "delimiter": self._delimiter,
+        }
+        if asset is not None:
+            if asset.container:
+                query_options["container"] = asset.container
+            if asset.name_starts_with:
+                query_options["name_starts_with"] = asset.name_starts_with
+            if asset.delimiter:
+                query_options["delimiter"] = asset.delimiter
+
+        path_list: List[str] = list_azure_keys(
+            azure=self._azure,
+            query_options=query_options,
+            recursive=False,
+        )
+        return path_list
+
+    def _get_full_file_path(
+        self,
+        path: str,
+        data_asset_name: Optional[str] = None,
+    ) -> str:
+        # Not used by this class but required to fulfill parent class
+        # Format: http://<storage_account_name>.blob.core.windows.net/<container_name>/<blob_name>
+        raise NotImplementedError()
```

Comment:
This method is never invoked by this class but is from the parent. S3 implements it but I don't believe it uses it in any capacity.

Reply:
@cdkini Please look at this interface method (and what it calls) in the following modules (in this order):
```
great_expectations/datasource/data_connector/file_path_data_connector.py
great_expectations/datasource/data_connector/configured_asset_file_path_data_connector.py
great_expectations/datasource/data_connector/inferred_asset_filesystem_data_connector.py
great_expectations/datasource/data_connector/configured_asset_s3_data_connector.py
great_expectations/datasource/data_connector/inferred_asset_s3_data_connector.py
great_expectations/datasource/data_connector/configured_asset_azure_data_connector.py
```
You will see that this method is always implemented, except that in certain cases it calls a method in the subclass, which has to be implemented.  In other words, `_get_full_file_path()` is an interface method at the `file_path_data_connector` level, but there is also the sub-method for getting the path from the data asset.

While at it, please also fix the typo (data_assert -> data_asset) in `configured_asset_s3_data_connector.py`.  Thanks!

Reply:
ah makes a lot of sense! I will implement the function with this next round of updates